File: cli/version-compose.ts
import fs from 'fs'
import path from 'path'

let filename = 'docker-compose-template.yml'
if (process.env.NEO4J_VERSION === '5') {
	filename = 'docker-compose-v5-template.yml'
}
if (process.env.OFFLINE === 'true') {
	filename = 'docker-compose-template-offline.yml'
	if (process.env.NEO4J_VERSION === '5') {
		filename = 'docker-compose-template-v5-offline.yml'
	}
}
let compose = fs.readFileSync(path.join(__dirname, '..', filename), { encoding: 'utf-8' })
compose = compose.replaceAll(/\$\{APP_NAME\}/g, process.env.APP_NAME)
compose = compose.replaceAll(/\$\{VERSION\}/g, process.env.npm_package_version)
compose = compose.replaceAll(/\$\{NEXT_PUBLIC_HOST\}/g, process.env.NEXT_PUBLIC_HOST)
compose = compose.replaceAll(/\$\{NEXT_PUBLIC_PREFIX\}/g, process.env.NEXT_PUBLIC_PREFIX)
compose = compose.replaceAll(/\$\{DOCKERHUB_NAME\}/g, process.env.DOCKERHUB_NAME)
fs.writeFileSync(path.join(__dirname, '..', `docker-compose.yml`), compose)

File: cli/version-compose.py
env = {
	"APP_NAME":'knowledge_graph',
	"VERSION":'2.2.9',
	"NEXT_PUBLIC_HOST":'http://localhost:3000',
	"DOCKERHUB_NAME":'maayanlab',
	"OFFLINE": False,
	"NEXT_PUBLIC_PREFIX": ""
}
with open('../.env') as o:
	for line in o:
		if line.strip() and not line.startswith("#"):
			key,val = line.strip().split("=")
			if key in env:
				print(key)
				env[key] = val

print(env)	
if env['APP_NAME'] == 'distillery':
	filename = '../docker-compose-template-distillery-offline.yml' if env['OFFLINE'] else '../docker-compose-template-distillery.yml'
else:	
	filename = '../docker-compose-template-offline.yml' if env['OFFLINE'] else '../docker-compose-template.yml'
with open(filename) as o:
	compose = o.read()
	compose = compose.replace('${APP_NAME}', env['APP_NAME'])
	compose = compose.replace('${VERSION}', env['VERSION'])
	compose = compose.replace('${NEXT_PUBLIC_HOST}', env['NEXT_PUBLIC_HOST'])
	compose = compose.replace('${NEXT_PUBLIC_PREFIX}', env['NEXT_PUBLIC_PREFIX'])
	compose = compose.replace('${DOCKERHUB_NAME}', env['DOCKERHUB_NAME'])
	print(compose)
	
with open('../docker-compose.yml', 'w') as w:
	w.write(compose)